src/decode.rs

```diff
@@ -4703,45 +4707,41 @@ unsafe fn decode_b(
                     f.cur_segmap,
                     f.b4_stride,
                 );
-                let diff: libc::c_uint = dav1d_msac_decode_symbol_adapt8(
+                let diff = dav1d_msac_decode_symbol_adapt8(
                     &mut ts.msac,
                     (ts.cdf.m.seg_id[seg_ctx as usize]).as_mut_ptr(),
-                    (8 as libc::c_int - 1 as libc::c_int) as size_t,
+                    (DAV1D_MAX_SEGMENTS - 1) as size_t,
                 );
-                let last_active_seg_id: libc::c_uint = (*f.frame_hdr)
+                let last_active_seg_id = (*f.frame_hdr)
                     .segmentation
                     .seg_data
-                    .last_active_segid as libc::c_uint;
-                b
-                    .seg_id = neg_deinterleave(
+                    .last_active_segid;
+
+                b.seg_id = neg_deinterleave(
                     diff as libc::c_int,
                     pred_seg_id as libc::c_int,
-                    last_active_seg_id.wrapping_add(1 as libc::c_int as libc::c_uint)
-                        as libc::c_int,
+                    last_active_seg_id + 1,
```

Comment:
Does `seg_id` increase by 1 generally?  It's not supposed to overflow at all, right?